### PR TITLE
WIP: DOMParser based implementation;

### PR DESCRIPTION
**Pull request by ondreian**
laying some groundwork and toying around with #18 

non of the parser events are wired up to the UI yet, they just print to the console, I didn't want to do any of that until I was sure of the new structure.

Writing to `SessionState` provides a good API boundary though, so I can experiment nicely here.

### CODE REVIEW

**Comment by chriscoyier** [comment]
I was just about to log a bug that Illthorn doesn't catch skill enhansives like StormFront does...

Screen Shot 2020-08-12 at 6 27 59 AM

But this update fixes it! 🍻

File: src/app/ui/hud/stance.js [hunk 1]
```diff
@@ -12,9 +12,10 @@ module.exports = class Stance {
   }
 
   view() {
-    const sess = Session.focused()
-    if (!sess) return
-    const stance = Lens.get(sess, "state.stance.text")
+    const stance = Lens.get(
+      Session.current,
+      "state.stance.text"
+    )
```
Comment (chriscoyier) [inline src/app/ui/hud/stance.js:18]: Is this file needed at all anymore? I see `vital.js` doesn't call it anymore and that seems fine if it can be simplified. Although the UI should probably list like "offensive", "defensive", "guarded" and all that rather than this internal ID.

Screen Shot 2020-08-16 at 2 40 45 PM

File: src/app/styles/_hilite.scss [hunk 2]
```diff
@@ -2,7 +2,7 @@
  * String highligting via :hilite 
  */
 
-.exist {
+a[exist] {
```
Comment (chriscoyier) [inline src/app/styles/_hilite.scss:5]: "The" and "a" seems to get classified with the `.monster` and then `.exists` is nested within, which ends up with weird highlighting:

CleanShot 2020-08-16 at 14 44 33@2x

We could fix that at the CSS level though easily, just wanna make sure that's intended.

File: src/app/styles/_hilite.scss [hunk 3]
```diff
@@ -28,7 +28,7 @@
   cursor: pointer;
   color: var(--link-color, #27a4fd);
 }
-.clickable .d {
+d {
```
Comment (chriscoyier) [inline src/app/styles/_hilite.scss:31]: `d` isn't a real HTML element, so maybe a span with a class instead? It looks very nice though:

Screen Shot 2020-08-16 at 2 47 01 PM

Is the idea that these ultimately clickable commands, where you can click it to do the command (in this case "go out"?)

**Comment by chriscoyier** [comment]
Talking is doubled up:

Screen Shot 2020-08-16 at 2 48 19 PM

On both sides:

Screen Shot 2020-08-16 at 2 48 38 PM

**Comment by chriscoyier** [comment]
Inventory seems doubled-up, both a vertical list and comma-separated list:

Screen Shot 2020-08-16 at 2 56 47 PM

**Comment by chriscoyier** [comment]
"look" in your own inventory doesn't return anything:

Screen Shot 2020-08-16 at 2 58 05 PM

**Comment by chriscoyier** [comment]
Known TODO: wire up streams.

**Comment by chriscoyier** [comment]
Round/Cast Time bars don't work. They are there, and they re-rerender when timers are triggered (I think?), but there are no values to use.

**Comment by chriscoyier** [comment]
There might be something weird with the Mana vital too, sometimes it would disappear for me.

File: src/app/styles/_hilite.scss (hunk 2 above)
Comment (ondreian) [reply src/app/styles/_hilite.scss:5]: Yeah, this is something that is inherent to the game feed itself, so I think I should stick to that as closely as possible and only fix "errors" from the feed

File: src/app/styles/_hilite.scss (hunk 3 above)
Comment (ondreian) [reply src/app/styles/_hilite.scss:31]: Yeah, `<d>` commands mean that the `innerText` is the command whereas some game feed elements declare a `d=` attribute for the command and the `innerText` is different.

**Comment by chriscoyier** [comment]
Sometimes game output text is within a roundtime element

```
illthorn:parser parsed:: +1ms <roundtime value="1597798318"><dialogdata class="expr"><label id="yourLvl" value="Level 67" top="0" left="0" align="n" width="160" height="15"><progressbar id="mindState" value="75" text="becoming numbed" top="45" left="3" align="n" width="160" height="15"><progressbar id="nextLvlPB" value="7" text="90579 until next level" top="20" left="3" align="n" width="160" height="15"></progressbar></progressbar></label></dialogdata>You search around and find 21 bloodscrip, which you pocket!</roundtime>

illthorn:parser buffer:: +1ms Roundtime: 5 sec.
```
